### PR TITLE
Feature/tt 2096

### PR DESCRIPTION
**Pull request by sarahentzel**

### CODE REVIEW

File: src/components/ScriptureTable.tsx [hunk 1]
```diff
@@ -718,7 +725,29 @@ export function ScriptureTable(props: IProps) {
...
+        cleanFileName(
+          (passageRec.attributes.book || '') + passageRec.attributes.reference
+        )
+      );
+      console.log(
+        'set fileName',
+        cleanFileName(
+          (passageRec.attributes.book || '') + passageRec.attributes.reference
+        )
+      );
```
Comment (gtryus) [inline src/components/ScriptureTable.tsx:746]: Did you intend to leave this here?

File: src/components/ScriptureTable.tsx [hunk 2]
```diff
@@ -1286,7 +1315,6 @@ export function ScriptureTable(props: IProps) {
           mediaRemoteIds[0]
       );
     }
-    setDoSave(true);
```
Comment (gtryus) [inline src/components/ScriptureTable.tsx:1289]: Not sure why we don't want to save after upload?

File: src/crud/useMediaAttach.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
-import { useGlobal } from 'reactn';
-import { ISharedStrings, ActivityStates, MediaFile } from '../model';
-import { orbitErr } from '../utils';
+import {useGlobal} from 'reactn';
+import {ISharedStrings,  ActivityStates,  MediaFile} from '../model';
+import { orbitErr} from '../utils';
```
Comment (gtryus) [inline src/crud/useMediaAttach.ts:3]: The spaces after curly brackets are not consistent. Here they are removed and lines 5-7 they are retained?

File: src/crud/useMediaAttach.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ interface IStateProps {
 interface IProps extends IStateProps, IDispatchProps {}
 
 export const useMediaAttach = (props: IProps) => {
-  const { doOrbitError, ts } = props;
+  const {doOrbitError, ts} = props;
```
Comment (gtryus) [inline src/crud/useMediaAttach.ts:19]: I think the spaces are supposed to be there, right?

File: src/crud/useMediaAttach.ts [hunk 5]
```diff
@@ -28,21 +28,21 @@ export const useMediaAttach = (props: IProps) => {
   ) => {
     var tb = new TransformBuilder();
     var ops: Operation[] = [];
+    var mediaRI = {type: 'mediafile', id: mediaId};
+    if (related(memory.cache.query(q => q.findRecord(mediaRI)), 'passage') === passage)
```
Comment (gtryus) [inline src/crud/useMediaAttach.ts:32]: this looks funny b/c in theory, media.cache.query might return null if the record is not found and we are not testing for it.

File: src/store/upload/actions.tsx [hunk 6]
```diff
@@ -36,12 +47,20 @@ export const writeFileLocal = (file: File, remoteName?: string) => {
   var fullName = local.localname;
   if (!remoteName && file.path === '') fullName += path.sep + file.name;
   createPathFolder(fullName);
+  while (fs.existsSync(fullName)) {
+    fullName = nextVersion(fullName);
+    console.log(fullName);
```
Comment (gtryus) [inline src/store/upload/actions.tsx:52]: Do we want to delete this line?

File: src/store/upload/actions.tsx [hunk 7]
```diff
@@ -36,12 +47,20 @@ export const writeFileLocal = (file: File, remoteName?: string) => {
...
+  }
   const reader = new FileReader();
   reader.onload = (evt) => {
-    fs.writeFileSync(fullName, evt?.target?.result, { encoding: 'binary' });
+    fs.writeFileSync(fullName, evt?.target?.result, {
+      encoding: 'binary',
+      flag: 'wx',
```
Comment (gtryus) [inline src/store/upload/actions.tsx:58]: perhaps add // write but fail if file exists

File: src/store/upload/actions.tsx [hunk 8]
```diff
@@ -36,12 +47,20 @@ export const writeFileLocal = (file: File, remoteName?: string) => {
...
+    });
   };
   reader.readAsBinaryString(file);
-  return path.join(PathType.MEDIA, file.name);
+  console.log(fullName.split(path.sep).pop());
```
Comment (gtryus) [inline src/store/upload/actions.tsx:62]: should this line be removed?

File: src/store/upload/actions.tsx [hunk 9]
```diff
@@ -68,6 +87,7 @@ export const nextUpload = (
     // offlineOnly
     try {
       var filename = writeFileLocal(files[n]);
+      console.log(filename);
```
Comment (gtryus) [inline src/store/upload/actions.tsx:90]: should this line be removed?

**Review by gtryus — CHANGES_REQUESTED** [review]
You can see the comments I made on various lines. You can merge after you address the comments. Let me know if you think I need to review again.

File: src/components/ScriptureTable.tsx (hunk 2 above)
Comment (sarahentzel) [reply src/components/ScriptureTable.tsx:1289]: Everything is saved as we go.

File: src/crud/useMediaAttach.ts (hunk 3 above)
Comment (sarahentzel) [reply src/crud/useMediaAttach.ts:3]: My ts and tsx files aren't being formatted correctly.  I've tried to figure out why but probably need to look at it together.  This was me attempting to fix what it did.

**Pull request merged**